packages/in_app_purchase/lib/src/google_play_connection.dart

```diff
@@ -0,0 +1,33 @@
+import 'dart:async';
+
+import 'billing_client_wrappers.dart';
+import 'in_app_purchase_connection.dart';
+
+/// An [InAppPurchaseConnection] that wraps Google Play Billing.
+///
+/// This translates various [BillingClient] calls and responses into the
+/// common plugin API.
+class GooglePlayConnection implements InAppPurchaseConnection {
+  BillingClient _billingClient = BillingClient();
+
+  @override
+  Future<bool> isAvailable() async {
+    return await _billingClient.isReady;
+  }
+
+  @override
+  Future<bool> connect() async {
```

Comment:
The docs on `InAppPurchaseConnection` says that this does nothing if the user is already connected, if that is the wanted behavior we probably shouldn't call `startConnection` if we're connected?